fetch.bs

```diff
@@ -6718,6 +6753,415 @@ agent's <a>CORS-preflight cache</a> for which there is a <a>cache entry match</a
+|
++ ---- + https://ok.example.com/good
+|      | 64 kibibytes, granted via the "{{PermissionsPolicy/deferred-fetch}}" policy.
+|      |
+|      + ---- + https://x.example.com
+|               0. Only documents with the same origin as the <a for=/>top-level traversable</a> can
+|               grant the 8 kibibytes based on the "{{PermissionsPolicy/deferred-fetch-minimal}}" policy.
+|
++ ---- + https://ok.example.com/redirect, navigated to https://x.example.com
+|        0. The reserved 64 kibibytes for https://ok.example.com are not available for https://x.example.com.
+|
++ ---- + https://ok.example.com/back, navigated to https://me.example.com
+         Shares quota with the <a for=/>top-level traversable</a>, as they're same origin.
+</code></pre>
+
+<p>In the above example, the <a for=/>top-level traversable</a> and its <a>same origin</a>
+descendants share a quota of 384 kibibytes. That value is computed as such:
+<ul>
+ <li><p>640 kibibytes are initially granted to the <a for=/>top-level traversable</a>.
+
+ <li><p>128 kibibytes are reserved for the "{{PermissionsPolicy/deferred-fetch-minimal}}" policy.
+
+ <li><p>64 kibibytes are reserved for the container navigating to
+ <code>https://ok.example/good</code>.
+
+ <li><p>64 kibibytes are reserved for the container navigating to
+ <code>https://ok.example/redirect</code>, and lost when it navigates away.
+
+ <li><code>https://ok.example.com/back</code> did not reserve 64 kibibytes, because it navigated
+ back to <a for=/>top-level traversable</a>'s origin.
+
+ <li><p>640 - 128 - 64 - 64 = 384 kibibytes.
+</ul>
+</div>
+
+
+<p>This specification defines a <a>policy-controlled feature</a> identified by the string
+"<dfn for=PermissionsPolicy enum-value>deferred-fetch</dfn>". Its
+<a for="policy-controlled feature">default allowlist</a> is "<code>self</code>".
+
+<p>This specification defines a <a>policy-controlled feature</a> identified by the string
+"<dfn for=PermissionsPolicy enum-value>deferred-fetch-minimal</dfn>". Its
+<a for="policy-controlled feature">default allowlist</a> is "<code>*</code>".
+
+<p>The <dfn>quota reserved for <code>deferred-fetch-minimal</code></dfn> is 128 kibibytes.
+
+<div algorithm>
+<p>To get the <dfn>available deferred-fetch quota</dfn> given a {{Document}}
+<var>controlDocument</var> and an <a for=/>origin</a>-or-null <var>origin</var>:
+
+<ol>
+ <li><p>Let <var>navigable</var> be <var>controlDocument</var>'s <a>node navigable</a>.
+
+ <li><p>Let <var>isTopLevel</var> be true if <var>controlDocument</var>'s <a>node navigable</a> is a
+ <a for=/>top-level traversable</a>; otherwise false.
+
+ <li><p>Let <var>deferredFetchAllowed</var> be true if <var>controlDocument</var> is
+ <a>allowed to use</a> the <a>policy-controlled feature</a> "{{PermissionsPolicy/deferred-fetch}}";
+ otherwise false.
+
+ <li><p>Let <var>deferredFetchMinimalAllowed</var> be true if <var>controlDocument</var> is
```

Comment:
> deferredFetchMinimalAllowed is true

Is it possible to bypass all these 5 conditions without a value for `quota`, i.e. `isTopLevel == false` && `deferredFetchAllowed == false` && `deferredFetchMinimalAllowed == false`?

Reply:
Yes, adding to the first condition.

Reply:
I think the latest matching conditions still don't cover `frame-4` below:

```
root (a.com) -> frame-1 (a.com) -> frame-2 (b.com)
             -> frame-3 (b.com) -> frame-4 (a.com)
```

`frame-4` has `isTopLevel == false`, `deferredFetchAllowed == false`, `deferredFetchMinimalAllowed == true`, but `reserved deferred-fetch quota` is 0, which leads to 0kb quota.

Reply:
Yea, that's right, it should receive 0.

Reply:
Added "otherwise" to the switch instead